### PR TITLE
ARXIVNG-2968: Improve AutoTeX log highlighting for "File Not Found" and "Emergency Stop" TeX errors.

### PR DESCRIPTION
**Pull request by DavidLFielding**
This ticket [ ARXIVNG-2968 ] was originally scoped with addressing the problem of submissions that fail but still seem to generate a PDF (bad).

The problem (or one problem) with the compiler service not signaling errors properly was fixed with recent changes to the converter/compilation image. I'm no longer able to generate a PDF given the example submission originally provided to demonstrate the issue.

The "File Not Found"  and the "Emergency Stop" errors are currently highlighted in the log body but not extracted and displayed in the log summary section.

In this PR I've added messages for the "File Not Found" and "Emergency Stop" errors to the log summary section. Since these two errors often occur together I only display one of the errors, the "File Not Found" error, when they occur together. We'll need to find another submission with an emergency stop to demonstrate the other message.

I've also improved the error message displayed when a user uploads XeTeX/LuaTeX source ( currently not supported).

Screen Shot 2020-03-05 at 4 21 22 PM

XeTeX/LuaTeX:

Screen Shot 2020-03-05 at 3 07 03 PM

I expect Shamsi might be the most interested in the visible log highlighting changes. I'm definitely interested in comments and suggestions to improve this feature of the process page.

Note: At the current time the highlighting routine is independent of the process HTML. That is, things discovered in the markup routine can't be communicated effectively to the processing step, such as to change the state of the submission. In the case where a submission compiles successfully, but we discover something really bad in the log highlighting, it would be nice to 'update' the status of the submission to failed status. Ideas for future improvements.

### CODE REVIEW

File: submit/filters/tex_filters.py [hunk 1]
```diff
@@ -17,6 +17,14 @@
 
 RUN_ORDER = ['last', 'first', 'second', 'third', 'fourth']
 
+def initialize_error_summary() -> str:
+    """Initialize the error_summary string with desired markuop."""
```
Comment (mhl10) [inline submit/filters/tex_filters.py:21]: Thinking about your other PR, should this return `Markup`?

File: submit/filters/tex_filters.py [hunk 2]
```diff
@@ -241,24 +249,35 @@ def compilation_log_display(autotex_log: str, submission_id: int,
...
+    # to address.
     error_summary = ''
+
+    # Keep track of any errors we've already added to error_summary
+    abort_markup = False
+    xetex_luatex_abort = False
+    emergency_stop = False
+    missing_file = False
     missing_font_markup = False
     rerun_markup = False
```
Comment (mhl10) [inline submit/filters/tex_filters.py]: Consider renaming some of these for extra clarity, per [decision on predicates](https://confluence.cornell.edu/pages/viewpage.action?pageId=372900900).

File: submit/filters/tex_filters.py [hunk 3]
```diff
@@ -355,80 +374,135 @@ def compilation_log_display(autotex_log: str, submission_id: int,
...
                     rerun_markup = True
 
+                # Missing file needs to be kicked up in visibility and displayed in
+                # compilation summary.
+                #
+                # There is an issue with the AutoTeX log where parts of the log
+                # may be getting truncated. Therefore, for this error, we will
+                # report the error if it occurs during any run.
+                #
+                # We might want to refine the activiation criteria for this
```
Comment (mhl10) [inline submit/filters/tex_filters.py:459]: Add `# TODO: ...` ?

**Review by mhl10 — APPROVED** [review]
Looks good, see minor feedback

**Comment by SBBCornell** [comment]
I tested the updated branch with the new converter version 1.5. It spins up successfully and returns the correct error message when a figure is missing! 

David, maybe we can discuss future tests and do a little planning. Im happy to keep testing as needed but feel my feedback will be most helpful with some parameters and specific goals.

File: submit/filters/tex_filters.py (hunk 2 above)
Comment (DavidLFielding) [reply submit/filters/tex_filters.py]: I will rename some of these predicates for clarity. This code was actually written prior to the 'decision on predicates'  decision but makes sense to update at this time.

**Pull request merged**